oemof_b3/tools/data_processing.py

```diff
@@ -1,7 +1,761 @@
+
+    # Make sure to only stack the DataFrame if it is not stacked already
+    stacked = False
+    for item in list(df.columns):
+        if item in required_header:
+            stacked = True
+
+    if not stacked:
+        # Convert timeindex column to datetime format
+        df["timeindex"] = pd.to_datetime(df[df.columns[0]])
+        # In case there is another datetime series with other header than timeindex,
+        # it is redundant and deleted
+        if df.columns[0] != "timeindex":
+            del df[df.columns[0]]
+        # Set timeindex as index
+        df = df.set_index("timeindex")
+
+        # Stack time series
+        df = stack_timeseries(df)
+
+    else:
+        # Read data with stacked time series out of a csv
+        df = pd.read_csv(path)
+
+        # Save header of DataFrame to variable
+        df_header = list(df.columns)
+
+        # Get file name
+        filename = os.path.splitext(path)[0]
+
+        # Check whether required columns are missing in the DataFrame
+        missing_required = []
+        for required in required_header:
+            if required not in df_header:
+                if "region" not in required:
+                    # Add required columns, that are missing, to a list
+                    missing_required.append(required)
+
+        # Interrupt if required columns are missing and print all affected columns
+        if len(missing_required) > 0:
+            raise KeyError(
+                f"The data in {filename} is missing the required column(s): {missing_required}"
+            )
+
+        # Set timeindex as default name of timeindex_start index
+        # This is necessary if DataFrame is to be unstacked afterwards
+        df["timeindex_start"].index.name = "timeindex"
+        # Convert to datetime format
+        df["timeindex_start"] = pd.to_datetime(df["timeindex_start"])
+        df["timeindex_stop"] = pd.to_datetime(df["timeindex_stop"])
+        # Convert series values from string to list
+        for number, item in enumerate(df["series"].values):
+            df["series"].values[number] = ast.literal_eval(item)
+
+    # "region" can be extraced from var_name. Therefore a further
+    # required header required_header_without_reg is introduced
+    required_header_without_reg = required_header.copy()
+    required_header_without_reg.remove("region")
+    # If optional columns are missing in the stacked DataFrame
+    if (
+        list(df.columns) == required_header
+        or list(df.columns) == required_header_without_reg
+    ) and list(df.columns) != header:
+        # ID in the form of numbering is added if "id_ts" is missing
+        if optional_header[0] not in df.columns:
+            df[optional_header[0]] = np.arange(0, len(df))
+
+        # The region is extracted out of "var_name"
+        if required_header[0] not in df.columns:
+            region = []
+            for row in np.arange(0, len(df)):
```

Comment:
Seems that the function relies on the data (Hard coded BE, BB). This should not be.

Reply:
But do we need other regions than these three?